compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -371,851 +139,31 @@ fn parse_args<'a>(
                         "positional arguments cannot follow named arguments",
                     );
                     err.span_label(e.span, "positional arguments must be before named arguments");
```

Comment:
What's the output for `format!("{x} {} {} {}", x = 1, 2, 3);`? I believe we'll have at least 2 errors for the non named args, right? [That's the current behavior](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c7e2cfc3df33a999f11a009a40341373). Should we instead try and accumulate all of these and emit a single error like

```
error: positional arguments cannot follow named arguments
 --> src/main.rs:2:36
  |
2 |     format!("{x} {} {} {}", x = 1, 2, 3);
  |                                 -  ^  ^ positional arguments must be before named arguments
  |                                 |
  |                                 named argument
```

Reply:
The current behavior is unchanged in this PR. (Except the span is now for the entire `x = 1` instead of just `1`.) We can improve it in a separate PR.